### PR TITLE
Allow deleting places/visits with invalid URLs

### PR DESCRIPTION
**Pull request by lyuyuan92**
#1206 
Tests done:
`cargo test --all`
`cargo fmt`
`cargo clippy --all-targets --all-features`
`cargo run -p sync-test`

### CODE REVIEW

File: components/places/src/storage/history.rs [hunk 1]
```diff
@@ -188,10 +188,15 @@ fn add_visit(
 
 /// Returns the GUID for the specified Url, or None if it doesn't exist.
 pub fn url_to_guid(db: &PlacesDb, url: &Url) -> Result<Option<SyncGuid>> {
+    string_to_guid(db, url.clone().into_string())
+}
+
+/// Returns the GUID for the specified Url String, or None if it doesn't exist.
+pub fn string_to_guid(db: &PlacesDb, url: String) -> Result<Option<SyncGuid>> {
```
Comment (thomcc) [inline components/places/src/storage/history.rs]: Nit: This should probably take a `&str`, and `url_to_guid` should use `url.as_str()`, (It should have already been doing this, but nobody noticed it during the review, presumably).

Maybe rename it to `href_to_guid` too (we use `href` which we use in a few places to refer to a string containing a url).

File: components/places/src/storage/history.rs [hunk 2]
```diff
@@ -233,8 +238,12 @@ pub fn delete_visits_between(db: &PlacesDb, start: Timestamp, end: Timestamp) ->
 }
 
 pub fn delete_place_visit_at_time(db: &PlacesDb, place: &Url, visit: Timestamp) -> Result<()> {
+    delete_str_place_visit_at_time(db, place.as_str(), visit)
+}
+
+pub fn delete_str_place_visit_at_time(db: &PlacesDb, place: &str, visit: Timestamp) -> Result<()> {
```
Comment (thomcc) [inline components/places/src/storage/history.rs]: Similarly, rename this to delete_place_visit_at_time_by_href` or something.

File: components/places/src/storage/history.rs (hunk 1 above)
Comment (lyuyuan92) [reply components/places/src/storage/history.rs]: Done

File: components/places/src/storage/history.rs (hunk 2 above)
Comment (lyuyuan92) [reply components/places/src/storage/history.rs]: Done

**Pull request merged**